dix/enterleave.c

```diff
@@ -629,6 +629,7 @@ FixDeviceValuator(DeviceIntPtr dev, deviceValuator * ev, ValuatorClassPtr v,
         ev->valuator0 = v->axisVal[first];
         break;
     default:
+        assert("ev->num_valuators becomes 0, it's a bug!");
```

Comment:
use BUG_ON() macro. it properly logs and shuts down the Xserver instead of just crashing it.